modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,425 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background for that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Add Dominant Color preload support to WordPress.
+ *
+ * TODO: Add tests
+ */
+class wp_Dominant_Color {
+
+
+	public function __construct() {
+
+		add_filter( 'wp_print_scripts', [ $this, 'add_styles' ] );
+		add_filter( 'wp_footer', [ $this, 'add_scripts' ] );
+
+		add_filter( 'wp_generate_attachment_metadata', [ $this, 'dominant_color_metadata' ], 10, 2 );
+		add_filter( 'wp_generate_attachment_metadata', [ $this, 'has_transparency_metadata' ], 10, 2 );
+
+		// do we have the new filter or are duplicating core the functions?
+		if ( has_filter( 'wp_img_tag_add_adjust' ) ) {
+			add_filter( 'wp_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 20, 3 );
+		} else {
+			add_filter( 'the_content', [ $this, 'filter_content_tags' ], 20 );
+			add_filter( 'wp_dominant_color_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 10, 3 );
+		}
+
+		add_filter( 'attachment_fields_to_edit', [ $this, 'add_dominant_color_media_setting' ], 10, 2 );
+		add_filter( 'attachment_fields_to_save', [ $this, 'save_dominant_color_media_setting' ], 10, 2 );
```

Comment:
I don't think we need the ability to save colour here. Let's generate this in the background for now and not at UI to edit it.